src/helpers/provider.ts

```diff
@@ -1,13 +1,16 @@
 import { StaticJsonRpcProvider } from '@ethersproject/providers';
 
-const providers = {};
+const providers: Record<number, StaticJsonRpcProvider | undefined> = {};
 
-export function getProvider(networkId: string | number) {
-  const url = `https://brovider.xyz/${networkId}`;
+export function getProvider(networkId: number): StaticJsonRpcProvider {
+  const url = `https://rpc.brovider.xyz/${networkId}`;
 
-  if (!providers[networkId]) {
-    providers[networkId] = new StaticJsonRpcProvider({ url, timeout: 25000 });
+  let provider = providers[networkId];
+
+  if (!provider) {
+    provider = new StaticJsonRpcProvider({ url, timeout: 25000 });
```

Comment:
I still can see request to `eth_chainId`, with this change it should avoid that
```suggestion
    provider = new StaticJsonRpcProvider({ url, timeout: 25000 }, networkId);
```